### PR TITLE
PartitionInducedCycleError: pass through CycleError

### PR DESCRIPTION
**Pull request by matthiasdiener**
Draft because:

- [ ] Needs https://github.com/inducer/pytools/pull/159
- [ ] Needs a pytools release

### CODE REVIEW

File: pytato/partition.py [hunk 1]
```diff
@@ -216,7 +216,9 @@ def make_partition(self, outputs: DictOfNamedArrays) -> GraphPartition:
                     pid_to_needing_pids,
                     lambda x: sorted(pid_to_output_names[x]))
         except CycleError as err:
-            raise PartitionInducedCycleError(err)
+            # type-ignore-reason: Needs a pytools release with
+            # https://github.com/inducer/pytools/pull/158 to pass mypy.
```
Comment (inducer) [inline pytato/partition.py:220]: I can do that. Remind me tomorrow.

File: pytato/partition.py [hunk 2]
```diff
@@ -216,7 +216,9 @@ def make_partition(self, outputs: DictOfNamedArrays) -> GraphPartition:
...
+            raise PartitionInducedCycleError(err)  # type: ignore[no-untyped-call]
```
Comment (inducer) [inline pytato/partition.py:221]: Maybe unpack the cycle rather than repackaging the whole error.

File: pytato/partition.py (hunk 1 above)
Comment (inducer) [reply pytato/partition.py:220]: https://pypi.org/project/pytools/2022.1.13/

File: pytato/partition.py (hunk 2 above)
Comment (matthiasdiener) [reply pytato/partition.py]: Is [4388cda](https://github.com/inducer/pytato/pull/390/commits/4388cda1a6ec4cc16dffb7891f4b477742a5d685) what you had in mind?

File: pytato/partition.py (hunk 1 above)
Comment (matthiasdiener) [reply pytato/partition.py]: Thanks, I removed the ignore in [1c7969f](https://github.com/inducer/pytato/pull/390/commits/1c7969f86541eb6ec4be7d7052b5084734936d9f)

**Comment by inducer** [comment]
This has conflicts now, sorry.

File: pytato/distributed/verify.py [hunk 3]
```diff
@@ -272,8 +272,8 @@ def add_needed_pid(pid: _DistributedPartId,
         from pytato.partition import PartitionInducedCycleError
         try:
             compute_topological_order(pid_to_needed_pids)
-        except CycleError:
-            raise PartitionInducedCycleError
+        except CycleError as err:
+            raise PartitionInducedCycleError(err.node)
```
Comment (inducer) [inline pytato/distributed/verify.py:276]: Based on https://github.com/inducer/pytools/pull/159, this should probably be `err.path`.

**Comment by inducer** [comment]
Converted to draft for the time being, with a to-do list in the description.

**Comment by inducer** [comment]
Superseded by #434 .

**Pull request closed without merging**